src/Definition/RuntimeDefinition.php

```diff
@@ -27,9 +27,7 @@ class RuntimeDefinition implements DefinitionInterface
      */
     public function __construct(?array $explicitClasses = null)
     {
-        if ($explicitClasses) {
-            $this->setExplicitClasses($explicitClasses);
-        }
+        $this->setExplicitClasses($explicitClasses ?? []);
```

Comment:
The type of  `private ?array $explicitClasses` should be refined a bit:

 1. should not be nullable
 2. should probably declare `array<class-string, bool>` (and this constructor adjusted accordingly, I think)

Reply:
Ah, that's even better. I'll have a closer look at this when I get home.

Reply:
@Ocramius I updated it, let me know what you think of it.
The only thing I'm wondering is the following: [extending clients](https://3v4l.org/Bj1Bl), in case someone did extend the previous case constructor with the same signature, passing null as default, this might lead to runtime crashes.

Reply:
We can do that in a separate patch: for now, this is a sufficient improvement, thanks! :+1: